core/foundation/inc/RtypesCore.h

```diff
@@ -55,7 +56,8 @@ typedef long           Long_t;      //Signed long integer 4 bytes (long)
 typedef unsigned long  ULong_t;     //Unsigned long integer 4 bytes (unsigned long)
 #endif
 typedef float          Float_t;     //Float 4 bytes (float)
-typedef float          Float16_t;   //Float 4 bytes written with a truncated mantissa
+typedef float          Float16_t;   //Float 4 bytes in memory, written as a 2 bytes float with a truncated mantissa
+typedef float          FloatHalf_t; //Float 4 bytes in memory, written as a 2 bytes float with IEEE half-precision
```

Comment:
If it is to be treated as `Double32_t`, we ougth to at least add it to:
https://root.cern.ch/doc/master/TClingUtils_8cxx_source.html#l00684
and to
`TDataType`

Reply:
Ah of course. Looking into it this seems to be enough for just cling support, if I'm not mistaken. Another question would then be if it makes sense to support it for the full I/O, but this can be further discussed and potentially addressed in a follow-up PR.